### PR TITLE
add signature keyring package and signed-by definition

### PR DESCRIPTION
**Pull request by azr4m**

### CODE REVIEW

**Comment by Yoda-BZH** [comment]
Shouldn't the package add a requirement on the keyring package ?

**Comment by azr4m** [comment]
Indeed, I set requirement for keyring package if on Debian Like OS

**Comment by Yoda-BZH** [comment]
`grains['os_family']` should cover more cases than listing a lot of OSes with `grains['os']`

**Comment by azr4m** [comment]
Indeed, I reused the code that was already done previously. I just pushed the modification with the use of the os_family grain

File: icinga2/init.sls [hunk 1]
```diff
@@ -1,24 +1,36 @@
 {% from "icinga2/map.jinja" import icinga2 with context %}
 
 
-{%- if grains['os'] in ['Debian', 'Ubuntu'] %}
+{%- if grains['os_family'] in ['Debian', 'Ubuntu'] %}
```
Comment (Yoda-BZH) [inline icinga2/init.sls]:
```suggestion
{%- if grains['os_family'] in ['Debian'] %}
```

File: icinga2/init.sls [hunk 2]
```diff
@@ -1,24 +1,36 @@
...
+    - name: deb [signed-by={{ icinga2.signed_by }}] https://{{ icinga2.repo }}/{{ salt['grains.get']('os_family')|lower }} icinga-{{ salt['grains.get']('oscodename') }} main
     - key_url: http://packages.icinga.com/icinga.key
     - file: /etc/apt/sources.list.d/icinga.list
-{%- elif grains['os'] == 'RedHat' %}
+    - clean_file: True
+{%- elif grains['os_family'] == 'RedHat' %}
 # TODO: RedHat repo info goes here
 {%- endif %}
 
+{%- if grains['os_family'] in ['Debian', 'Ubuntu'] %}
```
Comment (Yoda-BZH) [inline icinga2/init.sls]:
```suggestion
{%- if grains['os_family'] in ['Debian'] %}
```

File: icinga2/init.sls [hunk 3]
```diff
@@ -1,24 +1,36 @@
...
+  pkg.installed:
+    - name: {{ icinga2.keyring_package }}
+    - require:
+      - pkgrepo: icinga2_repo
+{%- endif %}
 
 icinga2_pkg:
   pkg.installed:
     - name: {{ icinga2.package }}
+{%- if grains['os_family'] in ['Debian', 'Ubuntu'] %}
```
Comment (Yoda-BZH) [inline icinga2/init.sls]:
```suggestion
{%- if grains['os_family'] in ['Debian'] %}
```

**Comment by Yoda-BZH** [comment]
Thank you for your contribution :)

**Pull request merged**